### ISSUE TITLE
TypeScript Definition File

### ISSUE THREAD

**Issue by ifiokjr** [googleapis/nodejs-tasks#140]
**Is your feature request related to a problem? Please describe.**

I've created TypeScript types for this repo with a PR currently available on [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/31145). 

Let me know if you'd prefer it within the repo and I can create a PR to add the types.

**Comment by JustinBeckwith** [comment]
Greetings @ifiokjr, and thanks for the types!  It's awesome that these are available on DefinitelyTyped.  I think it's fine to leave them there for now, while we figure out an all up plan to convert this library over to TypeScript.  Thanks for doing this!

**Comment by ifiokjr** [comment]
Sound great!

Also so you know, they haven't been merged yet so not currently available. It's being blocked by external dependencies on `google-gax` right now which is causing the build to fail.

**Comment by kirillgroshkov** [comment]
We need TypeScript types too

**Comment by grant** [comment]
Glad to find this issue via a Google Search!

I'm re-opening this issue because:
- I don't see the TypeScript Declaration File declared in `package.json`.
- I don't see any mention of the community TSD in this repo.
  - (https://www.npmjs.com/package/@types/google-cloud__tasks)

I'd expect this issue to be open to track the progress of official support.

---

For context, I'm building a Tasks application with TS and started searching around when I found no autocompletion for my methods.

**Comment by JustinBeckwith** [comment]
As of now, we had been tracking this over at https://github.com/googleapis/google-cloud-node/issues/952.  @grant - how would you prefer we track this?
- Keeping a single tracking issue over in google-cloud-node
- Letting people file them in each repo one at a time
- Both?

**Comment by grant** [comment]
Thanks for listening. I was sort of expecting a section like this when reading the README:

https://github.com/googleapis/google-api-nodejs-client#typescript

I understand the need to not have 80 duplicative issues, but even a:

> "Note: TypeScript definitions are currently _not_ supported and the feature request is being tracked in https://github.com/googleapis/google-cloud-node/issues/952"

...in each repo would be better than nothing or having developers like me look at GitHub issues.

**Comment by grant** [comment]
I must say as an anecdote the last few days, the lack of TypeScript definitions significantly decreases my productivity when creating API requests for this client.

I would use `googleapis` if the sample I was writing wasn't going to be as official.

**Comment by bcoe** [comment]
@grant as we move towards TypeScript micro-generators for gRPC libraries, we will be starting to introduce types.

**Comment by JustinBeckwith** [comment]
This was completed in `1.7.0` :)

**Comment by kirillgroshkov** [comment]
Sorry to resurrect an old issue, but it still doesn't export `types` in `package.json`. Are there plans to do so? Is there an issue to track this?

**Comment by JustinBeckwith** [comment]
No problem!  I'm not sure this is actually required, is it?  I just created a fresh project with the latest TypeScript module, `npm install`'d this library, and I was able to get Intellisense, and it compiled.

Are you running into specific problems compiling code or using the module?

**Comment by kirillgroshkov** [comment]
> No problem! I'm not sure this is actually required, is it? I just created a fresh project with the latest TypeScript module, `npm install`'d this library, and I was able to get Intellisense, and it compiled.
> 
> Are you running into specific problems compiling code or using the module?

Yes, my IntelliSense is working weirdly. Not completely broken, but also not as flawless as it does with modules that export types according to the spec (via `types`).

I do believe it's needed, or, at least, it's how TypeScript specifies it to be.

I know that it can infer it automatically if it's located at the root of the module and named `index.d.ts`. But since it's located in `build/src` - it's not inferred properly. That's my guess.

**Comment by kirillgroshkov** [comment]
https://www.typescriptlang.org/v2/docs/handbook/module-resolution.html#how-typescript-resolves-modules

**Comment by JustinBeckwith** [comment]
@alexander-fenster @bcoe this isn't a problem in my experiments, but maybe adding this explicitly to the `package.json` `types` field ain't a bad idea to just be double sure.

**Comment by Kezzsim** [comment]
I seem to be having a similar issue, might be because I'm new to typescript but I haven't been able to successfully overload or override types related to where Cloud Tasks intersect with App Engine requests.

<img width="1246" alt="Screen Shot 2020-12-18 at 11 37 05 AM" src="https://user-images.githubusercontent.com/15878154/102638783-0ac24a80-4126-11eb-9acd-2930ffd1851a.png">

Is there a type definition specifically for App Engine I can add to my dev-dependencies to suppress these? Otherwise I'll just keep trying to make my own interfaces with the same name and see if that works.

**Comment by bcoe** [comment]
@Kezzsim could I bother you to open an issue with the source you're running, and the error you're getting. To me it looks like types are being found, but that there's an issue with the expected parameters.

closing in favor of https://github.com/googleapis/gapic-generator-typescript/issues/804
